trestle/core/control_io.py

```diff
@@ -985,88 +1041,162 @@ def _comp_name_in_dict(comp_name: str, comp_dict: Dict[str, List[Dict[str, str]]
         label: str,
-        comp_dict: Dict[str, Dict[str, List[str]]],
+        comp_dict: CompDict,
         node: MarkdownNode,
         control_id: str,
-        comp_list: List[str]
+        comp_list: List[str],
+        component_mode=False
     ) -> None:
-        prose = ControlIOReader._clean_prose(node.content.text)
-        if node.key.startswith('### '):
-            if len(node.key.split()) <= 1:
-                raise TrestleError(f'Line in control {control_id} markdown starts with ### but has no component name.')
-            comp_name = node.key.split(' ', 1)[1].strip()
-            simp_comp_name = ControlIOReader.simplify_name(comp_name)
-            if simp_comp_name == ControlIOReader.simplify_name(const.SSP_MAIN_COMP_NAME):
-                raise TrestleError(
-                    f'Response in control {control_id} has {const.SSP_MAIN_COMP_NAME} as a component heading.  '
-                    'Instead, place all response prose for the default component at the top of th section, '
-                    'with no ### component specified.  It will be entered as prose for the default system component.'
-                )
-            if simp_comp_name in comp_list:
-                raise TrestleError(
-                    f'Control {control_id} has a section with two ### component headings for {comp_name}.  '
-                    'Please combine the sections so there is only one heading for each component in a statement.'
-                )
-            comp_list.append(simp_comp_name)
-            comp_name = ControlIOReader._comp_name_in_dict(comp_name, comp_dict)
-        if comp_name in comp_dict:
-            if label in comp_dict[comp_name]:
-                comp_dict[comp_name][label].extend(prose)
+        """Extract the label, prose, possible component name - along with implementation status."""
+        prose = '\n'.join(ControlIOReader._clean_prose(node.content.text))
+        if prose:
+            # for ssp, ### marks component name but for component it is ##
+            prefix = '## ' if component_mode else '### '
+            if node.key.startswith(prefix):
+                if len(node.key.split()) <= 1:
+                    raise TrestleError(
+                        f'Line in control {control_id} markdown starts with {prefix} but has no content.'
+                    )
+                comp_name = node.key.split(' ', 1)[1].strip()
+                simp_comp_name = ControlIOReader.simplify_name(comp_name)
+                if simp_comp_name == ControlIOReader.simplify_name(const.SSP_MAIN_COMP_NAME) and not component_mode:
+                    raise TrestleError(
+                        f'Response in control {control_id} has {const.SSP_MAIN_COMP_NAME} as a component heading.  '
+                        'Instead, place all response prose for the default component at the top of th section, with '
+                        'no ### component specified.  It will be entered as prose for the default system component.'
```

Comment:
I changed language a bit - but the goal is not to have a heading for the component name, like ### ACME, of the component name is This System.  For This System, it is the default name when no component name is entered.  This is all a bit confusing when it is trying to be helpful and clear.  I think it will be clear and beneficial once people understand it.  It has been how it works from the start of the SSP effort.